src/hooks/useSignChallenge.ts

```diff
@@ -0,0 +1,116 @@
+import { useState, useCallback } from 'react';
+import { useSignMessage } from 'wagmi';
+import { SIGNING_SERVICE_URL } from '../constants/constants';
+import { storageKeys } from '../constants/localStorage';
+import { SiweMessage } from 'siwe';
+
+export interface SiweSignatureData {
+  signatureSet: boolean;
+  expirationDate: string;
+}
+
+export function useSiweSignature(address?: `0x${string}`) {
+  const { signMessageAsync } = useSignMessage();
+  const [requiresSign, setRequiresSign] = useState(false);
```

Comment:
I would rename this and all references to it to 
```suggestion
  const [showSigning, setShowSigning] = useState(false);
```
or 
```suggestion
  const [signingPending, setSigningPending] = useState(false);
```

so that this indicates how the variable is supposed to be used (ie. for the modal). Otherwise it's confusing.